lib/LANraragi/Utils/Plugins.pm

```diff
@@ -114,25 +114,44 @@ sub get_plugin_parameters {
     my $plugin   = get_plugin($namespace);
     my %pluginfo = $plugin->plugin_info();
 
-    my @args = ();
+    my %args;
 
-    # Fill with default values first
-    foreach my $param ( @{ $pluginfo{parameters} } ) {
-        push( @args, $param->{default_value} );
-    }
+    if ( ref( $pluginfo{parameters} ) eq 'ARRAY' ) {
 
-    # Replace with saved values if they exist
-    if ( $redis->hexists( $namerds, "enabled" ) ) {
-        my $argsjson = $redis->hget( $namerds, "customargs" );
-        $argsjson = redis_decode($argsjson);
+        my @args;
```

Comment:
I would give this variable a different name, it's a bit confusing having a hash and an array with the same name in the scope of a single sub.